Assignment/Conversion.c

```diff
@@ -8,26 +8,32 @@
 
 #pragma warning(disable:4996)
 #include <stdio.h>
+
 void DecToBinary (int n);
 void DecToHexa (int n);
-
-// Function to convert the decimal number to binary
+int bitLength;
+/// <summary>Decimal number to binary Conversion</summary>
 void DecToBinary (int n) {
-   int binary[32];
-   int i = 0;
-   int flag = 0; // Flag is to check if the number is negative
-   if (n == 0)    printf ("binary is 0");
+   int binary[32], i = 0;
+   int isNegative = 0; // Flag is to check if the number is negative
+   if (n == 0)   printf ("0");  
    if (n < 0) {
       n = -n;
-      flag = 1;
+      isNegative = 1;
    }
    while (n > 0) {
       binary[i] = n % 2;
-      n = n / 2;
+      n /= 2;
       i++;

```

Comment:
Can give i++ in binary[i++] = n%2; This will increment the value of i after completing the expression.

Refer this for further information 
[https://www.geeksforgeeks.org/pre-increment-and-post-increment-in-c/](url)